src/client/js/main.jsx

```diff
@@ -1,14 +1,140 @@
-var Test = React.createClass({
-    render() {
-        return (
-            <div>{this.props.msg}</div>
-        );
-    }
+/*
+
+	Copied, pasted and edited code from react-router example. To refine later
+
+ */
+
+
+var { Route, DefaultRoute, RouteHandler, Link } = ReactRouter;
+
+var App = React.createClass({
+  getInitialState: function () {
+    return { memes: getAllMemes() };
+  },
+
+  // App's render function
+  render: function () {
+    
+  	// Set up links to other memes
+    var links = this.state.memes.map(function (meme) {
+      
+      return (
+        <li key={meme.name}>
+        
+          <Link
+
+            to="memes-detail"
+
+            params={{ name: meme.name.toLowerCase().replace(/\s/g, "_") }}
+          >
+          	{meme.name}
+          </Link>
+        </li>
+      );
+    });
+
+    // Return app's view to render function
+    return (
+      <div className="App">
+        <div className="Detail">
+          <RouteHandler/>
+        </div>
+        <ul className="Master">
+          {links}
+        </ul>
+      </div>
+    );
+  }
+});
+
+
+/* Index */
+var Index = React.createClass({
+  render: function () {
+    return <p>Select a meme below</p>;
+  }
 });
 
-var foo = 'babel@';
 
-React.render(
-    <Test msg={`React really works! Or does it? ${foo}!!!`} />,
-    document.getElementById('content')
-);
\ No newline at end of file
+/* Meme Class */
+var Meme = React.createClass({
+
+  contextTypes: {
+    router: React.PropTypes.func
+  },
+
+  imageUrl: function (name) {
+  	// Replace spaces in name with underscore
+  	var encodedName = name.replace(/\s/g, "_");
+
+  	// Return memes path with encoded name
+    return "../images/memes/" + encodedName + ".jpg";
```

Comment:
Great opportunity for an es6 template string here. Basically, with es6 enabled, we almost never have to do string concats again.